crates/astria-grpc-mock/src/mock_set.rs

```diff
@@ -0,0 +1,145 @@
+use std::{
+    ops::{
+        Index,
+        IndexMut,
+    },
+    sync::{
+        atomic::AtomicBool,
+        Arc,
+    },
+};
+
+use tokio::sync::Notify;
+use tracing::debug;
+
+use super::{
+    mock::Mock,
+    mounted_mock::MountedMock,
+};
+use crate::{
+    erase_request,
+    mounted_mock::MockResult,
+    verification::{
+        VerificationOutcome,
+        VerificationReport,
+    },
+};
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub(crate) enum MountedMockState {
+    InScope,
+    OutOfScope,
+}
+
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct MockId {
+    index: usize,
+}
+
+#[derive(Default)]
+pub(crate) struct MockSet {
+    mocks: Vec<(MountedMock, MountedMockState)>,
+}
+
+impl MockSet {
+    pub(crate) fn new() -> Self {
+        Self {
+            mocks: Vec::new(),
+        }
+    }
+
+    pub(crate) fn handle_request<
+        T: erased_serde::Serialize + prost::Name + Clone + Send + Sync + 'static,
+        U: Send + Sync + 'static,
+    >(
+        &mut self,
+        rpc: &'static str,
+        req: tonic::Request<T>,
+    ) -> tonic::Result<tonic::Response<U>> {
+        debug!("handling request.");
```

Comment:
remove or add more info?

Reply:
yeah, fair point. It's also reporting which rpc is being served. wiremock didn't have that because their requests are all dynamic, whereas ours aren't.